mat64/dense_arithmetic.go

```diff
@@ -361,9 +362,12 @@ func (m *Dense) Mul(a, b Matrix) {
 		panic(ErrShape)
 	}
 
-	var w Dense
+	var w *Dense
 	if m != a && m != b {
-		w = *m
+		w = m
+	} else {
+		m.reuseAs(ar, bc)
```

Comment:
I think you've misunderstood what this is doing.

The call the reuseAs here is not to allocate or resize, but to check if it can be reused. We know that m is non-zero if the operation is at all valid (a and b are non-zero), so the only thing that can be done in the reuseAs call is to pass the size checks or not. We panic in the second case and do nothing in the first case.

I'm confused about what you mean wrt the change in the inner test. The inner test (and the svd test) were previously incorrect; we had a nxm receiver failing to panic when calling an operation that produced a nx1 and nx1 not panicing when calling a 1x1 operation. Without the change I made in the tests, those tests would now panic. That actually threw me for a bit, because I was not expecting this fix to cause shape panics.